### PR TITLE
field-create: Make it easier for field types extending entity reference to add target type/bundles options (12.x)

### PR DESCRIPTION
**Pull request by DieterHolvoet**
Same as https://github.com/drush-ops/drush/pull/5556, but for 12.x.

### CODE REVIEW

File: src/Commands/field/FieldEntityReferenceHooks.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+namespace Drush\Commands\field;
+
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drush\Commands\DrushCommands;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class FieldEntityReferenceHooks extends DrushCommands
```
Comment (weitzman) [inline src/Commands/field/FieldEntityReferenceHooks.php:14]:
```suggestion
final class FieldEntityReferenceHooks extends DrushCommands
```
Lets add final unless we expect folks to extend this.

File: src/Commands/field/FieldEntityReferenceHooks.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    public static function create(ContainerInterface $container): static
+    {
+        return new static(
+            $container->get('entity_type.manager'),
+            $container->get('entity_type.bundle.info'),
+        );
+    }
+
+    /**
+     * @hook on-event field-create-field-storage
```
Comment (weitzman) [inline src/Commands/field/FieldEntityReferenceHooks.php]: these should be attributes

File: src/Commands/field/FieldEntityReferenceHooks.php (hunk 1 above)
Comment (DieterHolvoet) [reply src/Commands/field/FieldEntityReferenceHooks.php:14]: We do expect people to extend this, that's the whole point of this PR. This was explained in https://github.com/drush-ops/drush/pull/5556.

File: src/Commands/field/FieldEntityReferenceHooks.php (hunk 2 above)
Comment (DieterHolvoet) [reply src/Commands/field/FieldEntityReferenceHooks.php]: Do you know which attribute exactly? I can't seem to find an `OnEvent` attribute.

Comment (weitzman) [reply src/Commands/field/FieldEntityReferenceHooks.php]: https://github.com/drush-ops/drush/blob/0ddeed7a3f2ff5708135b7c9963194bb51fbfd4a/src/Drupal/Commands/sql/SanitizeCommentsCommands.php#L56
https://github.com/weitzman/drush/blob/0ddeed7a3f2ff5708135b7c9963194bb51fbfd4a/src/Drupal/Commands/sql/SanitizeCommentsCommands.php#L56

**Review by weitzman — APPROVED** [review]

**Pull request merged**